core-java-modules/core-java-string-conversions-2/src/test/java/com/baeldung/stringtocamelcase/StringtocamelcaseApplicationTests.java

```diff
@@ -0,0 +1,61 @@
+package com.baeldung.stringtocamelcase;
+
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class StringtocamelcaseApplicationTests {
+
+	private static final String TEXT_WHITESPACE = "THIS STRING SHOULD BE IN CAMEL CASE";
+	private static final String TEXT_UNDERSCORE = "THIS_STRING_SHOULD_BE_IN_CAMEL_CASE";
+	private static final String TEXT_EXPECTED = "ThisStringShouldBeInCamelCase";
+
+	@Test
+	public void testNothing() {
+	}
+
+	@Test
+	public void testToCamelCaseByIteration_shouldReturnCamelCase() {
+		assertEquals(TEXT_EXPECTED, StringtocamelcaseApplication.toCamelCaseByIteration(TEXT_WHITESPACE, ' '));
+		assertEquals(TEXT_EXPECTED, StringtocamelcaseApplication.toCamelCaseByIteration(TEXT_UNDERSCORE, '_'));
```

Comment:
Let's break each test case up. Also, while the constants may _dry out_ the code, they make the test hard to read itself. Better to write something like this:

```java
assertThat(toCamelCaseByIteration("CAMEL CASE ME", " "))
  .isEqualTo("CamelCaseMe");
```

Also note, though it's optional, that `assertThat` from AssertJ makes these tests easier to read because the actual is on the left, which is a more natural order.